numba/core/options.py

```diff
@@ -43,16 +43,19 @@ def set_flags(self, flags):
         if kws.pop('looplift', True):
             flags.set("enable_looplift")
 
-        if kws.pop('boundscheck', False):
-            flags.set("boundscheck")
-
         if kws.pop('_nrt', True):
             flags.set("nrt")
 
         if kws.pop('debug', config.DEBUGINFO_DEFAULT):
             flags.set("debuginfo")
             flags.set("boundscheck")
 
+        # This is checked for logically after 'debug' as it's possible to want
+        # to have debuginfo on from the kwarg debug=True but to have boundscheck
+        # off.
+        if 'boundscheck' in kws:
```

Comment:
`boundscheck` is defaulted to `False`. 
https://github.com/numba/numba/blob/9381206a03a3e7231d417f147474577745f6a720/numba/core/decorators.py#L27
This will always disable `boundscheck`